osquery/tables/utility/get_file.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <sys/stat.h>
+
+#include <boost/filesystem.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace tables {
+
+const std::map<fs::file_type, std::string> kTypeNames{
+    {fs::regular_file, "regular"},
+    {fs::directory_file, "directory"},
+    {fs::symlink_file, "symlink"},
+    {fs::block_file, "block"},
+    {fs::character_file, "character"},
+    {fs::fifo_file, "fifo"},
+    {fs::socket_file, "socket"},
+    {fs::type_unknown, "unknown"},
+    {fs::status_error, "error"},
+};
+
+#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__linux__)
+const char token_charset[] =
+  "0123456789"
+  "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+  "abcdefghijklmnopqrstuvwxyz";
+#else
+  const char token_charset[] =
+  "0123456789"
+  "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+#endif 
+
+std::string genFileToken(const size_t token_len) {
+  auto randchar = []() -> char {
+    const size_t max_index = (sizeof(token_charset) - 1);
+    return token_charset[rand() % max_index];
+  };
+
+  std::string token(token_len,0);
+  std::generate_n (token.begin(), token_len, randchar);
+
+  return token;
+}
+
+void genGetFileContent(const fs::path& path,
+                 const fs::path& parent,
+                 const std::string& pattern,
+                 QueryData& results) {
+  // Must provide the path, filename, directory separate from boost path->string
+  // helpers to match any explicit (query-parsed) predicate constraints.
+  struct stat file_stat, link_stat;
```

Comment:
A lot of this code is repeated from `readFile` and from `./osquery/tables/utility/file.cpp`, you can use other tables inline:
```cpp
auto results = SQL::selectAllFrom("file", "path", EQUALS, path.string());
```